src/session/__tests__/sessionReducer-test.js

```diff
@@ -38,4 +60,149 @@ describe('sessionReducer', () => {
     const newState = sessionReducer(baseState, eg.action.login_success);
     expect(newState).toEqual({ ...baseState, needsInitialFetch: true });
   });
+
+  test('DEAD_QUEUE', () => {
+    const state = deepFreeze({
+      ...baseState,
+      needsInitialFetch: false,
+      loading: true,
+    });
+    const newState = sessionReducer(state, deepFreeze({ type: DEAD_QUEUE }));
+    expect(newState).toEqual({ ...baseState, needsInitialFetch: true });
+  });
+
+  test('LOGOUT', () => {
+    const state = deepFreeze({
+      ...baseState,
+      lastNarrow: [],
+      needsInitialFetch: true,
+      loading: true,
+    });
+    const newState = sessionReducer(state, deepFreeze({ type: LOGOUT }));
+    expect(newState).toEqual(baseState);
+  });
```

Comment:
This looks reasonable; but when this test is first added, it tests not doing anything :slightly_smiling_face:

Reply:
Aha, and then when the change is made in the app code, the diff here is bigger. So that would be good to separate.

Looks like ACCOUNT_SWITCH is similar.